### ISSUE TITLE
NPC can't provide anything to help in form of items even with successful check 

### ISSUE THREAD

**Issue by vorpal-void** [cataclysmbnteam/Cataclysm-BN#1252]
# Describe the bug

When player asks NPC about any available job and there is one, it's always concludes in options about asking for advice and sharing equipment. TALK_SHARE_EQUIPMENT seems to me working as indented and provides necessary checks, but will always conclude in "NPC has nothing to give!" message (you can even thank them for it, nice)
In my experience, every NPC affected: the one that spawns in "fixed scenarios", any random encountered, and any traders from base game and/or from mods. See context below for additional details.

In short: right now NPC can't provide anything to help when you're accepting mission, but owe you literally everything when you complete it (most of the time). This is not very logical nor balanced.

# Steps To Reproduce

1. Start the game, create new world
2. Choose default "Play now!" fixed scenario
3. There will be NPC nearby, ask them for work
4. Convince them to share the stuff because you're helping them (it will most likely be 100% or close to it)
(you can repeat this check if failed - there is no practical punishment for asking multiple times in a row)
5. Thank them for nothing
6. Try nearby console to reveal refugee center
7. Debug teleport or walkthere
8. Repeat same stuff with trader who obviously HAS stuff to share
9. Thank them for nothing as well

# Expected behavior

NPC should be able to provide something practical to help player with accepted mission.

# Screenshot

![изображение](https://user-images.githubusercontent.com/60776130/148993332-0dd1fe76-4037-4c96-a0ad-b7972e2048bd.png)

# Versions and configuration

- OS: Windows 11
    - OS Version: 10.0 2009
- Game Version: 43e3981 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [dda]
]

# Additional context

Now things are getting interesting: they are, in fact, can share something - but only after successful completing mission. You can ask them to teach you something as a reward, or share equipment - and now this will work, because there is added debt for that NPC in form of mission value.

Said values for base game missions are out of hand, this is somewhat separate issue: they are too generous and this dialogue option is not a trial, but a fixed 100% success option that will provide player an option to simply take everything that NPC have as a reward because item values are far less than mission value is. Situation is a little better for refugee center since their value is less generous, but require a check as well since in some cases it provides opportunity to grab everything that NPC could provide in trade window, their Doctor as example. As continuation of steps above:

1. Debug "mark as complete" accepted mission from start-spawned NPC or complete it normally
2. Talk and ask for items as rewards.
3. Behold, they owe you a fortune that you're free to utilize right ahead or ask them later via "care to trade?" option

![изображение](https://user-images.githubusercontent.com/60776130/148994045-6022ff07-77c9-47d3-a0a5-bd221382efbc.png)

This is separate thing and "simple fix" via `missiondef.json` edit, but main issue here lies in the code that I'm not very good with; I think this should be something with `void talk_function::give_equipment( npc &p )` in `npctalk_funcs.cpp`.
